eth/downloader/downloader.go

```diff
@@ -554,7 +554,7 @@ func (d *Downloader) syncWithPeer(p *peerConnection, hash common.Hash, td, ttd *
 			origin = 0
 		} else {
 			pivotNumber := pivot.Number.Uint64()
-			if pivotNumber <= origin {
+			if pivotNumber > 0 && pivotNumber <= origin {
```

Comment:
This is odd. If we're in this clause, it hemas that `height` is non-zero. Whih means that `latest` is non-zero. So basically, we're using genesis as pivot-block even though our own latest is more recent. 

Fixing this here seems like a bandaid, would be good to instead figure out the root cause of this. In what cases have you experienced this?